odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -310,6 +311,10 @@ def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
         if partner_ids & child_ids:
             raise UserError(_("You cannot merge a contact with one of his parent."))
 
+        # check if the list of partners to merge are linked to more than one user
+        if Users.with_context(active_test=False).search_count([('partner_id', 'in', partner_ids.ids)]) > 1:
```

Comment:
we will fetch user_ids below, so it will save us 1 query (and it makes the code shorter)

```suggestion
        if len(partner_ids.user_ids) > 1:
```